#!/usr/bin/env python
PACKAGE = "flir_gige"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("fps", double_t, 0, "frame per second", 20, 1, 60)
gen.add("raw", bool_t, 0, "Raw 14-bit data", False)
gen.add("nuc_action", bool_t, 0, "Non-uniform image correction", False)
nuc_enum = gen.enum([gen.const("nuc_manual", int_t, 0, "Manual"),
                     gen.const("nuc_automatic", int_t, 1, "Automatic")],
                     "An enum to set nuc mode")
gen.add("nuc_mode", int_t, 0, "Non-uniform correction mode", 0, 0, 1, edit_method=nuc_enum)
exit(gen.generate(PACKAGE, "flir_gige", "FlirGigeDyn"))
